File: GolangStudy/Arithmetic/insert_sort.go
func insertionSort(arr []int) []int {
	for i := range arr{
		preIndex := i -1 
		current := arr[i]
		for preIndex >= 0 && arr[preIndex] > current{
			arr[preIndex+1] = arr[preIndex]
			preIndex -= 1
		}
		arr[preIndex+1] = current
	}
	return arr
}

File: PHPStudy/Arithmetic/insert_sort.php
<?php

function insertionSort($arr){
    $len = count($arr);
    for($i = 1; $i < $len; $i++){
        $preIndex = $i - 1;
        $current = $arr[$i];
        while($preIndex >= 0 && $arr[$preIndex] > $current){
            $arr[$preIndex+1] = $arr[$preIndex];
            $preIndex--;
        }
        $arr[$preIndex+1] = $current;
    }
    return $arr;
}